packages/components/src/toggle-group-control/toggle-group-control-option-base/component.tsx

```diff
@@ -111,18 +141,39 @@ function ToggleGroupControlOptionBase(
 						<ButtonContentView>{ children }</ButtonContentView>
 					</button>
 				) : (
+					// @ts-expect-error Even if we're passing `button` through the
+					// `render` prop, the component is still expecting `input` props
 					<Radio
 						{ ...commonProps }
-						{
-							...otherContextProps /* these are only for Ariakit Radio */
-						}
-						as="button"
+						onFocus={ (
+							event: React.FocusEvent< HTMLInputElement >
+						) => {
+							// @ts-expect-error Even if we're passing `button` through the
+							// `render` prop, the component is still expecting `input` props
+							commonProps.onFocus?.( event );
+							if ( event.defaultPrevented ) return;
+							toggleGroupControlContext.setValue( value );
+						} }
+						render={ <button /> }
```

Comment:
You can pass the `onFocus` prop directly to `<button />` to get the correct types.

Reply:
Good point, done in https://github.com/WordPress/gutenberg/pull/50278/commits/2f3e1e77086647f3bfdaa5cbfef520886be792a5